data/mods/roulettemons/random-teams.ts

```diff
@@ -0,0 +1,140 @@
+/* eslint max-len: ["error", 240] */
+
+import {PRNG, PRNGSeed} from '../../../sim/prng';
+import RandomTeams from '../../random-teams';
+
+export class RandomRoulettemonsTeams extends RandomTeams {
+	constructor(format: Format | string, prng: PRNG | PRNGSeed | null) {
+		super(format, prng);
+		this.noStab = [...this.noStab, 'gigaimpact'];
+	}
+
+	getPokemonPool(
+		type: string,
+		pokemonToExclude: RandomTeamsTypes.RandomSet[] = [],
+		isMonotype = false,
+	) {
+		const pokemonPool = [];
+		const roulettemons = [
+			'koatric', 'aquazelle', 'salamalix', 'brawnkey', 'stuneleon', 'chillyte',
+			'eartharoo', 'crazefly', 'electritar', 'aquatopus', 'scorpita', 'baloon',
+			'kinesel', 'glacida', 'pidgeotine', 'gorilax', 'albatrygon', 'chillvark',
+			'komodith', 'giranium', 'flamyle', 'voltecta', 'ostria', 'ninjoth', 'herbigator',
+			'anteros', 'gladiaster', 'hyperoach', 'barracoth', 'toados', 'voltarak',
+			'mosqung', 'flamepion', 'hyenix', 'rhinolite', 'bellena', 'falcola', 'beanium',
+			'lemotic', 'biceon', 'skeleray', 'specyte', 'ramron', 'panthee', 'blastora',
+			'balar', 'dropacle', 'fluffora', 'dolphena', 'tigire', 'catelax', 'chillytemega',
+		];
```

Comment:
Why? You already list all the roulettemons in the formats-data file; why not use `dex.species.all()` and just filter out Pokémon without `randomBattleMoves`?

Reply:
Vanilla mons still had random battle moves (did I check the wrong place?) but filtering for 'heightm' gave the right list of mons.